test/strategies/elkilpstrategy.js

```diff
@@ -0,0 +1,308 @@
+const {expect} = require("chai")
+const {ethers, run} = require("hardhat")
+const staking_contract = require("../resources/abis/StakingRewardsILPV2.json");
+const {BigNumber} = ethers
+
+const MAX_UINT = "115792089237316195423570985008687907853269984665640564039457584007913129639935";
+
+describe("elk_ilp_strategy_v5", function () {
+
+    const elk_pair = require("../resources/abis/ElkPair.json");
+    const elk_router = require("../resources/abis/ElkRouter.json");
+
+    before(async () => {
+        await run("compile")
+    })
+
+    let owner
+    let account1
+    let account2
+    const timelockAddress = "0x8d36C5c6947ADCcd25Ef49Ea1aAC2ceACFff0bD7"
+    const depositTokenAddress = "0x2612dA8fc26Efbca3cC3F8fD543BCBa72b10aB59"
+    const elkTokenAddress = "0xE1C110E1B1b4A1deD0cAf3E42BfBdbB7b5d7cE1C"
+    const wavaxTokenAddress = "0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7"
+    const elkRouterAddress = "0x9E4AAbd2B3E60Ee1322E94307d0776F2c8e6CFbb"
+    const elpStakingRewardAddress = "0xA811738E247b27ec3C82873b4273425b5355bd71"
+    
+    let wavaxTokenContract
+    let elkTokenContract
+    let elkRouterContract
+    let elkPairContract
+    let stakingContract
+    let elkIlpStrategyV5
+    beforeEach(async () => {
+        let accounts = await ethers.getSigners()
+        owner = accounts[0]
+        account1 = accounts[1]
+        account2 = accounts[2]
+        
+        elkPairContract = await ethers.getContractAt(elk_pair, depositTokenAddress);
+        elkRouterContract = await ethers.getContractAt(elk_router, elkRouterAddress);
+        stakingContract = await ethers.getContractAt(staking_contract, elpStakingRewardAddress);
+        const elkILPStrategyV5Factory = await ethers.getContractFactory("ELKILPStrategyV5")
+        elkIlpStrategyV5 = await elkILPStrategyV5Factory.deploy(
+            "ElkIlpStrategy-Test",
+            depositTokenAddress, // Deposit token is the ELK-WAVAX liquidity pool address
+            elkTokenAddress, // Address for ELK Token
+            elpStakingRewardAddress, // Address of the StakingRewardsILPV2 for ELK-WAVAX farm
+            wavaxTokenAddress, // Address of swapPair token 1 (WAVAX),
+            elkTokenAddress, // Address of swapPair token 0 (ELK)
+            timelockAddress, // Timelock contract for YY
+            1,
+            1,
+            1,
+            1
+        )
+        await elkIlpStrategyV5.deployed()
+        wavaxTokenContract = await ethers.getContractAt("IWAVAX", wavaxTokenAddress)
+        elkTokenContract = await ethers.getContractAt("IWAVAX", elkTokenAddress)
+        //makes sure owner has enough WAVAX balance
+        if ((await wavaxTokenContract.balanceOf(owner.address)).lt("1000000000000000000000")) {
```

Comment:
would be nice to extract those into a utility function, those are repeated and can be reutilized by a lot